servers/audio/effects/audio_effect_capture.cpp

```diff
@@ -0,0 +1,158 @@
+/*                       This file is part of:                           */
+/*                           GODOT ENGINE                                */
+/*                      https://godotengine.org                          */
+/*************************************************************************/
+/* Copyright (c) 2007-2021 Juan Linietsky, Ariel Manzur.                 */
+/* Copyright (c) 2014-2021 Godot Engine contributors (cf. AUTHORS.md).   */
+/*                                                                       */
+/* Permission is hereby granted, free of charge, to any person obtaining */
+/* a copy of this software and associated documentation files (the       */
+/* "Software"), to deal in the Software without restriction, including   */
+/* without limitation the rights to use, copy, modify, merge, publish,   */
+/* distribute, sublicense, and/or sell copies of the Software, and to    */
+/* permit persons to whom the Software is furnished to do so, subject to */
+/* the following conditions:                                             */
+/*                                                                       */
+/* The above copyright notice and this permission notice shall be        */
+/* included in all copies or substantial portions of the Software.       */
+/*                                                                       */
+/* THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,       */
+/* EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF    */
+/* MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.*/
+/* IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY  */
+/* CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,  */
+/* TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE     */
+/* SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.                */
+/*************************************************************************/
+
+#include "audio_effect_capture.h"
+
+bool AudioEffectCapture::can_get_buffer(int p_frames) const {
+	return buffer.data_left() >= p_frames;
+}
+
+PackedVector2Array AudioEffectCapture::get_buffer(int p_frames) {
+	ERR_FAIL_COND_V(!buffer_initialized, PackedVector2Array());
+	ERR_FAIL_COND_V(p_frames >= buffer.size(), PackedVector2Array());
+	ERR_FAIL_COND_V(p_frames <= 0, PackedVector2Array());
+	int data_left = buffer.data_left();
+	if (data_left < p_frames) {
+		return PackedVector2Array();
+	}
+
+	PackedVector2Array ret;
+	ret.resize(p_frames);
+
+	Vector<AudioFrame> streaming_data;
+	streaming_data.resize(p_frames);
+	buffer.read(streaming_data.ptrw(), p_frames);
+	for (int32_t i = 0; i < p_frames; i++) {
+		ret.write[i] = Vector2(streaming_data[i].l, streaming_data[i].r);
+	}
+	return ret;
+}
+
+void AudioEffectCapture::clear_buffer() {
+	const int32_t data_left = buffer.data_left();
+	buffer.advance_read(data_left);
+}
+
+void AudioEffectCapture::_bind_methods() {
```

Comment:
Can you add the documentation for the new API? You can generate the template with `--doctool .` and if relevant use markup described in https://docs.godotengine.org/en/latest/community/contributing/class_reference_writing_guidelines.html#doc-class-reference-writing-guidelines